packages/flutter_tools/bin/xcode_backend.dart

```diff
@@ -0,0 +1,466 @@
+        echoError('You can fix this by updating the LOCAL_ENGINE environment variable, or');
+        echoError('by running:');
+        echoError('  flutter build ios --local-engine=ios_$buildMode');
+        echoError('or');
+        echoError('  flutter build ios --local-engine=ios_${buildMode}_unopt');
+        echoError('========================================================================');
+        exitApp(-1);
+      }
+      flutterFramework = '${environmentEnsure('FLUTTER_ENGINE')}/out/$localEngine/Flutter.xcframework';
+    }
+    String bitcodeFlag = '';
+    if (environment['ENABLE_BITCODE'] == 'YES' && environment['ACTION'] == 'install') {
+      bitcodeFlag = 'true';
+    }
+
+    // TODO(jmagman): use assemble copied engine in add-to-app.
+    if (existsDir('$projectPath/.ios')) {
+      runSync(
+        'rsync',
+        <String>[
+          '-av',
+          '--delete',
+          '--filter',
+          '- .DS_Store',
+          flutterFramework,
+          '$derivedDir/engine',
+        ],
+        verbose: verbose,
+      );
+    }
+
+    final List<String> flutterArgs = <String>[];
+
+    if (verbose) {
+      flutterArgs.add('--verbose');
+    }
+
+    if (environment['FLUTTER_ENGINE'] != null && environment['FLUTTER_ENGINE']!.isNotEmpty) {
+      flutterArgs.add('--local-engine-src-path=${environment['FLUTTER_ENGINE']}');
+    }
+
+    if (environment['LOCAL_ENGINE'] != null && environment['LOCAL_ENGINE']!.isNotEmpty) {
+      flutterArgs.add('--local-engine=${environment['LOCAL_ENGINE']}');
+    }
+
+    flutterArgs.addAll(<String>[
+      'assemble',
+      '--no-version-check',
+      '--output=${environment['BUILT_PRODUCTS_DIR'] ?? ''}/',
+      '-dTargetPlatform=ios',
+      '-dTargetFile=$targetPath',
+      '-dBuildMode=$buildMode',
+      '-dIosArchs=${environment['ARCHS'] ?? ''}',
+      '-dSdkRoot=${environment['SDKROOT'] ?? ''}',
+      '-dSplitDebugInfo=${environment['SPLIT_DEBUG_INFO'] ?? ''}',
+      '-dTreeShakeIcons=${environment['TREE_SHAKE_ICONS'] ?? ''}',
+      '-dTrackWidgetCreation=${environment['TRACK_WIDGET_CREATION'] ?? ''}',
+      '-dDartObfuscation=${environment['DART_OBFUSCATION'] ?? ''}',
+      '-dEnableBitcode=$bitcodeFlag',
+      '--ExtraGenSnapshotOptions=${environment['EXTRA_GEN_SNAPSHOT_OPTIONS'] ?? ''}',
+      '--DartDefines=${environment['DART_DEFINES'] ?? ''}',
+      '--ExtraFrontEndOptions=${environment['EXTRA_FRONT_END_OPTIONS'] ?? ''}',
+    ]);
+
+    if (environment['PERFORMANCE_MEASUREMENT_FILE'] != null && environment['PERFORMANCE_MEASUREMENT_FILE']!.isNotEmpty) {
+      flutterArgs.add('--performance-measurement-file=${environment['PERFORMANCE_MEASUREMENT_FILE']}');
+    }
+
+    final String? expandedCodeSignIdentity = environment['EXPANDED_CODE_SIGN_IDENTITY'];
+    if (expandedCodeSignIdentity != null && expandedCodeSignIdentity.isNotEmpty && environment['CODE_SIGNING_REQUIRED'] != 'NO') {
+      flutterArgs.add('--dCodesignIdentity=$expandedCodeSignIdentity');
```

Comment:
```suggestion
      flutterArgs.add('-dCodesignIdentity=$expandedCodeSignIdentity');
```
I just tried this:
```
Xcode's output:
↳
    ♦ /Users/magder/Projects/flutter/bin/flutter --verbose assemble
    --no-version-check
    --output=/Users/magder/Projects/test_create/build/ios/Release-iphoneos/
    -dTargetPlatform=ios -dTargetFile=lib/main.dart -dBuildMode=release
    -dIosArchs=arm64 armv7
    -dSdkRoot=/Users/magder/Applications/Xcode-12-5_beta3.app/Contents/Developer
    /Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS14.5.sdk
    -dSplitDebugInfo= -dTreeShakeIcons=true -dTrackWidgetCreation=false
    -dDartObfuscation=false -dEnableBitcode= --ExtraGenSnapshotOptions=
    --DartDefines= --ExtraFrontEndOptions=
    --dCodesignIdentity=3BF526324380ACCF26E8C3F18C0EAD79EAF5CA8E
    release_ios_bundle_flutter_assets
    [  +85 ms] executing: sysctl hw.optional.arm64
    [  +18 ms] Exit code 1 from: sysctl hw.optional.arm64
    [        ] sysctl: unknown oid 'hw.optional.arm64'
    [   +5 ms] executing: [/Users/magder/Projects/flutter/] git -c
    log.showSignature=false log -n 1 --pretty=format:%H
    [   +8 ms] Exit code 0 from: git -c log.showSignature=false log -n 1
    --pretty=format:%H
    [        ] 52243985501e340d06628b1dddbe4764fbf08eee
    [        ] executing: [/Users/magder/Projects/flutter/] git tag --points-at
    52243985501e340d06628b1dddbe4764fbf08eee
    [  +15 ms] Exit code 0 from: git tag --points-at
    52243985501e340d06628b1dddbe4764fbf08eee
    [   +1 ms] executing: [/Users/magder/Projects/flutter/] git describe --match
    *.*.* --long --tags 52243985501e340d06628b1dddbe4764fbf08eee
    [  +19 ms] Exit code 0 from: git describe --match *.*.* --long --tags
    52243985501e340d06628b1dddbe4764fbf08eee
    [        ] 2.4.0-4.0.pre-159-g5224398550
    [   +9 ms] executing: [/Users/magder/Projects/flutter/] git rev-parse
    --abbrev-ref --symbolic @{u}
    [   +6 ms] Exit code 128 from: git rev-parse --abbrev-ref --symbolic @{u}
    [        ] fatal: upstream branch 'refs/heads/refactor-xcode-backend' not
    stored as a remote-tracking branch
    [   +5 ms] executing: [/Users/magder/Projects/flutter/] git rev-parse
    --abbrev-ref HEAD
    [   +7 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
    [        ] refactor-xcode-backend
    [  +14 ms] Running shutdown hooks
    [        ] Shutdown hooks complete
    [   +3 ms] exiting with code 64

    [  +40 ms] Could not find an option named "dCodesignIdentity".
```
This would have failed in post-submit since devicelab has codesigning set up.

Reply:
this fails for me locally now, i wonder why it didn't yesterday.

Reply:
fixed and unit test updated to cover